### PR TITLE
Tests for idler which doesn't watch all namespaces

### PR DESCRIPTION
**Pull request by alexeykazakov**
Tests for https://github.com/codeready-toolchain/member-operator/pull/204

TODO:

- [x] Add tests for other controllers (ReplicaSet, DemploymentConfig, etc). Not only for Deployments.

Update: See https://github.com/codeready-toolchain/toolchain-e2e/pull/174

### CODE REVIEW

File: test/e2e/user_workloads_test.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	for _, p := range podsToIdle {
+		err := s.memberAwait.WaitUntilPodDeleted(p.Namespace, p.Name)
+		require.NoError(s.T(), err)
+	}
+
+	// Noise pods are still there
+	labels := make(map[string]string)
+	labels["app"] = "idler-hello-openshift"
+	_, err = s.memberAwait.WaitForPods(idlerNoise.Name, labels, len(podsNoise), wait.UntilPodRunning())
+	require.NoError(s.T(), err)
```
Comment (xcoulon) [inline test/e2e/user_workloads_test.go:64]: could you also check the result to make sure that the number of pods is correct?

File: test/e2e/user_workloads_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+	// Wait for the pods to be deleted
+	for _, p := range podsToIdle {
+		err := s.memberAwait.WaitUntilPodDeleted(p.Namespace, p.Name)
+		require.NoError(s.T(), err)
+	}
+
+	// Noise pods are still there
+	labels := make(map[string]string)
+	labels["app"] = "idler-hello-openshift"
```
Comment (xcoulon) [inline test/e2e/user_workloads_test.go:62]:
```suggestion
	labels := map[string]string{"app": "idler-hello-openshift"}
```

File: test/e2e/user_workloads_test.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	_, err = s.memberAwait.WaitForPods(idlerNoise.Name, labels, len(podsNoise), wait.UntilPodRunning())
+	require.NoError(s.T(), err)
+}
+
+func (s *userWorkloadsTestSuite) prepareWorkloads(idler *v1alpha1.Idler) []*corev1.Pod {
+	// Create a Deployment with two pods
+	var err error
+	replicas := int32(2)
+	labels := make(map[string]string)
+	labels["app"] = "idler-hello-openshift"
```
Comment (xcoulon) [inline test/e2e/user_workloads_test.go:72]:
```suggestion
	labels := map[string]string{"app": "idler-hello-openshift"}
```

File: test/e2e/user_workloads_test.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+func (s *userWorkloadsTestSuite) SetupSuite() {
+	s.ctx, s.hostAwait, s.memberAwait = WaitForDeployments(s.T(), &v1alpha1.UserSignupList{})
+	s.setApprovalPolicyConfig("automatic")
+}
+
+func (s *userWorkloadsTestSuite) TearDownTest() {
+	s.ctx.Cleanup()
+}
+
+func (s *userWorkloadsTestSuite) TestIdler() {
```
Comment (xcoulon) [inline test/e2e/user_workloads_test.go:35]:
```suggestion
func (s *userWorkloadsTestSuite) TestIdlerOnDeployment() {
```

File: test/e2e/user_workloads_test.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+	}
+
+	// Noise pods are still there
+	labels := make(map[string]string)
+	labels["app"] = "idler-hello-openshift"
+	_, err = s.memberAwait.WaitForPods(idlerNoise.Name, labels, len(podsNoise), wait.UntilPodRunning())
+	require.NoError(s.T(), err)
+}
+
+func (s *userWorkloadsTestSuite) prepareWorkloads(idler *v1alpha1.Idler) []*corev1.Pod {
```
Comment (xcoulon) [inline test/e2e/user_workloads_test.go:67]:
```suggestion
func (s *userWorkloadsTestSuite) prepareDeploymentWorkloads(idler *v1alpha1.Idler) []*corev1.Pod {
```

Comment (xcoulon) [reply test/e2e/user_workloads_test.go:67]: so there is room for other kinds of workloads (replicaset, standalone pods, etc.)

File: test/e2e/user_workloads_test.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+	idler, err := s.memberAwait.WaitForIdler("test-idler-dev")
+	require.NoError(s.T(), err)
+
+	// Noise
+	idlerNoise, err := s.memberAwait.WaitForIdler("test-idler-stage")
+	require.NoError(s.T(), err)
+
+	// Create payloads for both users
+	podsToIdle := s.prepareWorkloads(idler)
+	podsNoise := s.prepareWorkloads(idlerNoise)
```
Comment (xcoulon) [inline test/e2e/user_workloads_test.go:47]:
```suggestion
	podsToIdle := s.prepareDeploymentWorkloads(idler)
	podsNoise := s.prepareDeploymentWorkloads(idlerNoise)
```

Comment (xcoulon) [reply test/e2e/user_workloads_test.go:47]: see suggestion below

File: test/e2e/user_workloads_test.go (hunk 1 above)
Comment (alexeykazakov) [reply test/e2e/user_workloads_test.go:64]: WaitForPods() already does it

Comment (xcoulon) [reply test/e2e/user_workloads_test.go:63]: ok

File: test/e2e/user_workloads_test.go (hunk 4 above)
Comment (alexeykazakov) [reply test/e2e/user_workloads_test.go:35]: I was planning to use this function for other payloads too.

**Comment by alexeykazakov** [comment]
Closed in favor of https://github.com/codeready-toolchain/toolchain-e2e/pull/174

**Pull request closed without merging**